tests/baseline/tsfresh_features_test.py

```diff
@@ -0,0 +1,417 @@
+
+    Running the test
+    ================
+
+    .. code-block:: bash
+
+        cd "<YOUR_TSFRESH_DIR>"
+        python -m pytest tests/baseline/tsfresh_features_test.py
+
+
+    Test the test fails
+    ===================
+
+    To test that the test fails as desired and as does what it is supposed to do
+    there are 2 methods to achieve this:
+
+    - Modify the first value in your local tsfresh/examples/data/test_tsfresh_baseline_dataset/data.json
+      and run the test, then delete the modified local data.json file to be pulled down again.
+    - Modify a feature name or value in your local tests/baseline/tsfresh-<TSFRESH_BASELINE_VERSION>.py<PYTHON_VERSION>.data.json.features.transposed.csv file,
+      run the test and either pull it again or revert the change
+
+    """
+    def setUp(self):
+        self.baseline_ts_json = download_json_dataset()
+
+    def tearDown(self):
+        # Remove the directory after the test
+        try:
+            shutil.rmtree(test_path)
+        except:
+            pass
+
+    def test_tsfresh_baseline(self):
+        """
+        This test should generates a set of resources from a sample 60 data
+        point timeseries, extracts the features and saves a transposed csv of
+        feature_name,value that is used to compare against to baseline
+        transposed csv
+        """
+        baseline_ts_json = self.baseline_ts_json
+
+        test_path_created = False
+        if os.path.exists(test_path):
+            test_path_created = True
+        self.assertEqual(test_path_created, True)
+
+        json_data_exists = False
+        if os.path.isfile(baseline_ts_json):
+            json_data_exists = True
+
+        self.assertEqual(json_data_exists, True)
+
+        timeseries_json = None
+        if os.path.isfile(baseline_ts_json):
+            with open(baseline_ts_json, 'r') as f:
+                timeseries_json = json.loads(f.read())
+
+        if python_version == 2:
+            timeseries_str = str(timeseries_json).replace('{u\'results\': ', '').replace('}', '')
+        if python_version == 3:
+            timeseries_str = str(timeseries_json).replace('{\'results\': ', '').replace('}', '')
+
+        full_timeseries = eval(timeseries_str)
+        timeseries = full_timeseries[:60]
+        self.assertEqual(int(timeseries[0][0]), 1369677886)
+        self.assertEqual(len(timeseries), 60)
+
+        for ts, value in timeseries:
+            metric = 'tsfresh_features_test'
+            timestamp = int(ts)
+            value = str(float(value))
+            utc_ts_line = '%s,%s,%s\n' % (metric, str(timestamp), value)
+            with open(tmp_csv, 'a') as fh:
+                fh.write(utc_ts_line)
+
+        if os.path.isfile(tmp_csv):
+            file_created = True
+        else:
+            file_created = False
+        self.assertEqual(file_created, True)
+
+        df_features = None
+        if os.path.isfile(t_fname_out):
+            return True
```

Comment:
This was an artefact from from me trying to get the data to only be created once by the unittest, removed.